lib/puppet/provider/cisco_bgp/nxapi.rb

```diff
@@ -200,11 +199,6 @@ def timer_bestpath_limit_set
     @bgp_vrf.timer_bestpath_limit_set(limit, always)
   end
 
-  def confederation_peers_set
-    return unless @property_flush[:confederation_peers]
-    @bgp_vrf.confederation_peers_set(@property_flush[:confederation_peers])
-  end
-
   def flush
     if @property_flush[:ensure] == :absent
       @bgp_vrf.destroy
```

Comment:
I expected to see this in self.instances:

```
current_state[:confederation_peers] = obj.confederation_peers
```
Did you get it to work without doing that? Also, I expected to see an explicit getter/setter (see network or redistribute).